specs/derivation.md

```diff
@@ -940,6 +1049,232 @@ entries.
 
 [deposit-contract-spec]: deposits.md#deposit-contract
 
+### Network upgrade automation transactions
+
+[network upgrade automation transactions]: #network-upgrade-automation-transactions
+
+Some network upgrades require automated contract changes or deployments at specific blocks.
+To automate these, without adding persistent changes to the execution-layer,
+special transactions may be inserted as part of the derivation process.
+
+#### Ecotone
+
+The Ecotone hardfork activation block, contains the following transactions in this order:
+
+- L1 Attributes Transaction, using the pre-Ecotone `setL1BlockValues`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - GasPriceOracle deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+  - Update GasPriceOracle Proxy ERC-1967 Implementation Slot
+  - GasPriceOracle Enable Ecotone
+  - Beacon block roots contract deployment (EIP-4788)
+
+To not modify or interrupt the system behavior around gas computation, this block will not include any sequenced
+transactions by setting `noTxPool: true`.
+
```

Comment:
We need to specify the first `setL1BlockValuesEcotone` call to hold the invariant that the view of the system config matches what is on L1. Per https://github.com/ethereum-optimism/optimism/pull/8723, the current scalar value should be migrated to the `baseFeeScalar` and the `blobBaseFeeScalar` should be 0